agent.go

```diff
@@ -787,6 +788,16 @@ func (s *sandbox) listenToUdevEvents() {
 	}
 }
 
+func (s *sandbox) runOOMEventMonitor(ch <-chan struct{}, containerID string) {
+	go func() {
+		for {
+			<-ch
```

Comment:
not sure, but I think we are leaking routines here, what happen when the container is restarted? `runOOMEventMonitor` is called again and a new routine for the new container is started, but what happen with the old routines ?

Reply:
Yes, I think: first , the container's oomCh could be closed when the container terminated,
then check the channel, if it's closed, break thhis loop and exit this go routine.

Reply:
Yep I can add logic to see if the channel is closed and then break the routine.

Reply:
I've updated it to check if the channel is closed, and if so it will break the routine. This should prevent leaking of goroutines if the channel is closed.